flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriteRequestExecutorImpl.java

```diff
@@ -51,27 +64,46 @@ class ChannelStateWriteRequestExecutorImpl implements ChannelStateWriteRequestEx
     private final Thread thread;
     private volatile Exception thrown = null;
     private volatile boolean wasClosed = false;
-    private final String taskName;
+
+    private final Map<SubtaskID, BlockingQueue<ChannelStateWriteRequest>> unreadyQueues =
+            new ConcurrentHashMap<>();
+
+    private final JobID jobID;
+    private final Set<SubtaskID> subtasks;
+    private final AtomicBoolean isRegistering = new AtomicBoolean(true);
```

Comment:
There is `BlockingQueue`, two `volatile`, `ConcurrentHashMap` and `AtomicBoolean`. I think that's a bit too much and it should be all replaced with a single `synchronized (this)` or `private finla Object lock = new Object();` and later `synchronized (lock)` combined with non-thread safe queue, Exception, boolean and Map.